spring-cloud-sleuth-core/src/main/java/org/springframework/cloud/sleuth/trace/DefaultTracer.java

```diff
@@ -125,6 +136,7 @@ public Span close(Span span) {
 			}
 			SpanContextHolder.close();
 		}
+		this.spanDurationReporterService.submitDuration(span.getName(), span.getAccumulatedMillis());
```

Comment:
oh ic.. there's a double here.. maybe better to switch getAccumulatedMillis a unit like micros so we can deal in whole numbers.